code/__HELPERS/icons.dm

```diff
@@ -1184,28 +1184,37 @@ GLOBAL_LIST_EMPTY(transformation_animation_objects)
 		animate(pixel_x = initialpixelx + rand(-pixelshiftx,pixelshiftx), pixel_y = initialpixely + rand(-pixelshifty,pixelshifty), time = shake_interval)
 	animate(pixel_x = initialpixelx, pixel_y = initialpixely, time = shake_interval)
 
-///Checks if the given iconstate exists in the given file, caching the result. Setting scream to TRUE will print a stack trace ONCE.
-/proc/icon_exists(file, state, scream)
+/// Checks whether a given icon state exists in a given icon file. If `file` and `state` both exist,
+/// this will return `TRUE` - otherwise, it will return `FALSE`.
+///
+/// If you want a stack trace to be output when the given state/file doesn't exist, use
+/// `/proc/icon_exists_or_scream()`.
+/proc/icon_exists(file, state)
 	var/static/list/icon_states_cache = list()
-	if(icon_states_cache[file]?[state])
-		return TRUE
+	if(isnull(file) || isnull(state))
+		return FALSE //This is common enough that it shouldn't panic, imo.
 
-	if(icon_states_cache[file]?[state] == FALSE)
-		return FALSE
+	if(isnull(icon_states_cache[file]))
+		icon_states_cache[file] = list()
+		for(var/istate in icon_states(file))
+			icon_states_cache[file][istate] = TRUE
 
-	var/list/states = icon_states(file)
+	var/does_it_exist = !isnull(icon_states_cache[file][state])
+	return does_it_exist
```

Comment:
```suggestion
	return !isnull(icon_states_cache[file][state])
```

Reply:
I actually considered this - but I felt `return !isnull(icon_states_cache[file][state])` was a bit hard to understand. That's why I put the variable there.

Do you still want that suggested change applied?

Reply:
The isnull() makes it easy to know what the intention is. Small changes like this should not effect your readability

Reply:
It shouldn't - but I feel it does.

That's why I'm waiting for a maintainer's call here.

Reply:
Actually, what am I saying. The purpose of the return value is already obvious from the function itself - it's `TRUE` if the icon exists, and `FALSE` if it doesn't.

So I'm applying this change.